### ISSUE TITLE
"Is it okay to only use Train_GCC-training.tsv and Validation_GCC-1.1.0-Validation.tsv the from CC3M during training?"

### ISSUE THREAD

**Issue by HuMillion** [Pter61/context_i2w#19]
img2dataset --url_list Train_GCC-training.tsv --input_format "tsv"\ --url_col "url" --caption_col "caption" --output_format webdataset\ --output_folder /tym/datasets/cc_3M/train/ --processes_count 16 --thread_count 64 --image_size 256\ --enable_wandb True

i only find  Train_GCC-training.tsv  Validation_GCC-1.1.0-Validation.tsv, not   ├── Train_GCC-training_output.csv ## training data list
                                                                                                                          └── Validation_GCC-1.1.0-Validation_output.csv ## validation data list

**Comment by Pter61** [comment]
Yes, it’s fine to use only `Train_GCC-training.tsv` and `Validation_GCC-1.1.0-Validation.tsv` from CC3M during training. 

The important point is that these files contain the image URLs and captions required to download the actual images and captions for training.

In the `img2dataset` command you shared, you’re correctly using these `.tsv` files to download and preprocess the data:

```bash
img2dataset --url_list Train_GCC-training.tsv --input_format "tsv" \
--url_col "url" --caption_col "caption" --output_format webdataset \
--output_folder /tym/datasets/cc_3M/train/ --processes_count 16 --thread_count 64 \
--image_size 256 --enable_wandb True
```

This will download the images and captions to your specified folder and format them for training. You don’t need the `.csv` output files (`Train_GCC-training_output.csv` and `Validation_GCC-1.1.0-Validation_output.csv`) unless you want to inspect them separately or for some additional processing.

Since CC3M only provides image-text pairs (no triplets like reference-modified-target in CIR datasets), using either the training or validation set (or both) is fine for tasks that require standard image-text pairs. In my experiments, I typically only use the training set (`Train_GCC-training.tsv`) for training the mapping network, as it already contains sufficient data.
